Sources/SwiftDocC/Semantics/Metadata/Availability.swift

```diff
@@ -91,15 +101,15 @@ extension Metadata {
         public var platform: Platform
 
         /// The platform version that this page applies to.
-        @DirectiveArgumentWrapped
-        public var introduced: String
+        @DirectiveArgumentWrapped(name: .custom("introduced"))
+        public var introducedVersion: VersionTriplet
```

Comment:
That's a good discussion point, thanks for bringing it up.

I think we could potentially leave this variable as `introducedVersion` without renaming it, so that we only need to remove `introduced` after 6.1 is released, WDYT? 

If you feel strongly about not leaving this property as `introducedVersion` I'm happy to revert it back to a source-breaking change

Reply:
I just saw we now have `deprecated` as well - d6e121722725d7c29dc4d51f422ee658839ec53f

Might we worth making the source breaking change in this PR for both of them to be `VersionTriplet`s

Reply:
> I just saw we now have `deprecated` as well - [d6e1217](https://github.com/apple/swift-docc/commit/d6e121722725d7c29dc4d51f422ee658839ec53f)
> 
> Might we worth making the source breaking change in this PR for both of them to be `VersionTriplet`s

Yes, I think both properties should be some form of version type. However, I'm not sure what version type that should be. 

After looking at more of DocC's code I think find at least these types being used:
- [`SemanticVersion`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Model/Rendering/SemanticVersion.swift#L14) which looks like a copy of  `SymbolKit.SymbolGraph.SemanticVersion`
- [`Version`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Utility/Version.swift#L12)
- [`VersionTriplet`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Infrastructure/External%20Data/ExternalMetadata.swift#L14)
- [`Platform.Version`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Indexing/Navigator/AvailabilityIndex%2BExt.swift#L209)

Long term I think we should strive to get rid of this duplication to one version type per repo, but for this PR we only need to pick which version type we want to use going forward.

Reply:
FWIW: I think the deprecated property should continue to be called `deprecated`. This leads to an inconsistency with the `introducedVersion` property, so I think it's worth doing a second rename after the next release to have both properties be named consistently again.

Reply:
> Yes, I think both properties should be some form of version type. However, I'm not sure what version type that should be.
> 
> After looking at more of DocC's code I think find at least these types being used:
> 
> * [`SemanticVersion`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Model/Rendering/SemanticVersion.swift#L14) which looks like a copy of  `SymbolKit.SymbolGraph.SemanticVersion`
> * [`Version`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Utility/Version.swift#L12)
> * [`VersionTriplet`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Infrastructure/External%20Data/ExternalMetadata.swift#L14)
> * [`Platform.Version`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Indexing/Navigator/AvailabilityIndex%2BExt.swift#L209)
> 
> Long term I think we should strive to get rid of this duplication to one version type per repo, but for this PR we only need to pick which version type we want to use going forward.

I don't have a strong opinion on which type we settle on going forward, they all seem to be fulfilling a similar purpose. I have the most familiarity with  [`VersionTriplet`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Infrastructure/External%20Data/ExternalMetadata.swift#L14) because that's the type which is used for determining whether a platform is in beta, but I'm not familiar with the others or how extensively they're used throughout the codebase.

Opened: https://github.com/apple/swift-docc/issues/970

Reply:
I propose that we don't try to decide which type we choose going forward here, as I think that requires evaluation, and leave that decision to https://github.com/apple/swift-docc/issues/970

Either way that task will require a lot of deprecation / source-breaking changes, so I don't think the risk is increased by a lot by not making a decision here.

Reply:
Based on a quick search, `SemanticVersion` seems to be the most widely used type in DocC so far.

Reply:
I've made the change to `SemanticVersion` in commits 7efe5c196ad2280374a36b7b1aadfb2fdac2787e and 7d729c0e39b05366ea0501d350a430df67096580, WDYT?

Heads up that this PR is back to being a source-breaking change per our discussion to avoid having to do multiple renames in 6.1 and 6.2.